pkg/provider/azure.go

```diff
@@ -239,6 +239,11 @@ type Config struct {
 	// DisableAvailabilitySetNodes disables VMAS nodes support when "VMType" is set to "vmss".
 	DisableAvailabilitySetNodes bool `json:"disableAvailabilitySetNodes,omitempty" yaml:"disableAvailabilitySetNodes,omitempty"`
 
+	// DisableAzureStackCloud disables AzureStackCloud support. It should be used
+	// when setting AzureAuthConfig.Cloud with "AZURESTACKCLOUD" to customize ARM endpoints
+	// while the cluster is not running on AzureStack.
+	DisableAzureStackCloud bool `json:"disableAzureStackCloud,omitempty" yaml:"disableAzureStackCloud,omitempty"`
```

Comment:
how can I apply this in CSI driver, we use `az.InitializeCloudFromSecret()` or `azure.NewCloudWithoutFeatureGates(f)` to get cloud object: 
https://github.com/kubernetes-sigs/azuredisk-csi-driver/blob/master/pkg/azuredisk/azure.go#L58

Reply:
you could set `disableAzureStackCloud` in secret data part together with other cloud config options.

Reply:
or, another option is setting cloud.config.DisableAzureStackCloud out side of the secret

Reply:
why not add a new parameter in `InitializeCloudFromSecret` and `NewCloudWithoutFeatureGates`,
`InitializeCloudFromSecret` read secret, while `NewCloudWithoutFeatureGates` read file, we have 3 CSI drivers at least, it's better change as little as possible in CSI driver side.

Reply:
Azure Disk CSI driver already has a flag `--support-azure-stack`, so if these two funcs have a new paramter(`supportAzureStack`), that's minimum change.

Also what about change in AKS rp setting for `azure-cloud-controller-manager`, and `azure-cloud-node-manager`? Shall we disable by default or only disable in AKS rp?

Reply:
example from Azure Disk CSI driver:
```
        - name: azuredisk
          image: mcr.microsoft.com/k8s/csi/azuredisk-csi:latest
          imagePullPolicy: IfNotPresent
          args:
            ...
            - "--support-azure-stack=true"
```

shall we also add a similar flag for `azure-cloud-controller-manager` and `azure-cloud-node-manager`, that's easier for configure.

Reply:
cloud provider is not using InitializeCloudFromSecret now, hence more changes are needed if we change InitializeCloudFromSecret.

And the cloud name is setting in cloud config file (actually all azure specific configures are inside config file). Hence, it would be consistent when setting this new config together.

What's the overhead of setting this config inside secret? as what I understand, it would need no code changes besides vendoring the new library.